packages/augur-api/src/api/Events.ts

```diff
@@ -5,39 +5,48 @@ import {abi} from "@augurproject/artifacts";
 import {Abi} from "ethereum";
 
 export class Events {
-    private readonly provider: Provider;
-    private readonly augurAddress: string;
+  private readonly provider: Provider;
+  private readonly augurAddress: string;
 
-    public constructor(provider: Provider, augurAddress: string) {
-        this.provider = provider;
-        this.augurAddress = augurAddress;
-        this.provider.storeAbiData(<Abi>abi["Augur"], "Augur");
-    }
+  public constructor(provider: Provider, augurAddress: string) {
+    this.provider = provider;
+    this.augurAddress = augurAddress;
+    this.provider.storeAbiData(<Abi>abi["Augur"], "Augur");
+  }
 
-    public async getLogs(eventName: string, fromBlock: number, toBlock: number, additionalTopics?: Array<string | Array<string>>): Promise<Array<ParsedLog>> {
-        let topics: Array<string | Array<string>> = [this.provider.getEventTopic("Augur", eventName)];
-        if (additionalTopics) {
-            topics = topics.concat(additionalTopics);
-        }
-        const logs = await this.provider.getLogs({fromBlock, toBlock, topics, address: this.augurAddress});
-        return this.parseLogs(logs);
+  public async getLogs(
+    eventNames: Array<string>,
+    fromBlock: number,
+    toBlock: number,
+    additionalTopics?: Array<string | Array<string>>
+  ): Promise<Array<Log>> {
+    let events: Array<string> = [];
+    for (let eventName of eventNames) {
+      events.push(this.provider.getEventTopic("Augur", eventName));
     }
-
-    public parseLogs(logs: Log[]): ParsedLog[] {
-        return _.map(logs, (log) => {
-            const logValues = this.provider.parseLogValues("Augur", log);
-            return Object.assign(
-                logValues,
-                {
-                    blockNumber: log.blockNumber,
-                    blockHash: log.blockHash,
-                    transactionIndex: log.transactionIndex,
-                    removed: log.removed,
-                    transactionLogIndex: log.transactionLogIndex,
-                    transactionHash: log.transactionHash,
-                    logIndex: log.logIndex,
-                }
-            )
-        });
+    let topics: Array<string | Array<string>> = [events];
+    if (additionalTopics) {
+      topics = topics.concat(additionalTopics);
     }
+
+    return this.provider.getLogs({fromBlock, toBlock, topics, address: this.augurAddress});
+  }
+
+  public parseLogs(logs: Array<Log>): Array<ParsedLog> {
+    return _.map(logs, (log) => {
```

Comment:
I believe the TypeScript Array type has a map method.

Reply:
Oh, you're right! We're using use lodash's map function in a few other places in the augur-state code, and we don't need to be. Will change this.